README.md

```diff
@@ -180,6 +176,27 @@ Then run:
 This will generate 2 files:
 ```classes_-p.png``` and ```packages_-p.png``` which then can be added as images to the C4 "Code diagram" tab.
 
+## Azure setup
+The app runs on the Azure Cloud. Azure hosts the production application, in a docker container, as an Azure
+App Service, and using Azure's CosmosDB, which has an API compatible with MongoDB.
+
+Ensure you have an Azure account created at the [Azure portal](https://portal.azure.com)
+
+* Setup 
+    * Step 1: Follow the instructions to [Install azure CLI](https://docs.microsoft.com/en-us/cli/azure/) on your machine if you haven't already.
+    * Step 2: Add resource group: ```az group create -l uksouth -n <resource_group_name>```
+    * Step 3: Setup the Cosmos Database with Mongo API: 
+        * Create a new CosmosDB: ```az cosmosdb create --name <cosmos_account_name> --resourcegroup <resource_group_name> --kind MongoDB```
+        * Create new MongoDB database under that account: ```az cosmosdb mongodb database create --account-name <cosmos_account_name> --name <database_name> --resourcegroup <resource_group_name>```
+    * Step 4: Connect the app to the CosmosDB: ```az cosmosdb keys list -n <cosmos_account_name> -g <resource_group_name> --type connection-strings```. This will give us a connection string. Alter the given connection string to tell pymongo to use the default database: Add ```/DefaultDatabase``` after the port number in the given connection string, eg: ```mongodb://<database_name>:<primarymasterkey>@<database_name>.mongo.cosmos.azure.com:10255/DefaultDatabase?ssl=true&<config>```. Keep this connection string as we will encrypt in Step 7
+    * Step 5: Restrict access to the database: Find the (IP address for the web app)[https://docs.microsoft.com/en-us/azure/app-service/overview-inbound-outbound-ips#find-outbound-ips], and restrict the Cosmos DB to only allowing access from that IP via by using a [firewall](https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-configure-firewall).
+    * Step 6: Create a web app: 
+        * First create an App Service Plan: ```az appservice plan create --resource-group <resource_group_name> -n <appservice_plan_name> --sku Free --is-linux```
+        * Then create the Web App: ```az webapp create --resource-group <resource_group_name> --plan <appservice_plan_name> --name <webapp_name> --deployment-container-image-name <dockerhub_username>/todoapp:latest```
+    * Step 7: Setup the environment variables from your ```.env``` file: e.g ```az webapp config appsettings set -g <resource_group_name> -n <webapp_name> --settings FLASK_APP=todo_app/app```. Or you can pass in a JSON file containing all variables by using ```--settings @foo.json```, see [here](https://docs.microsoft.com/en-us/cli/azure/webapp/config/appsettings?view=azure-cli-latest#az_webapp_config_appsettings_set). Ensure you include the ```DB_CONNECTION_STRING``` which should be the string retrived in step 5.
```

Comment:
This all sounds reasonable, but where did you create it?
Your [project exercise resource group](https://portal.azure.com/#@softwireacademy.onmicrosoft.com/resource/subscriptions/d33b95c7-af3c-4247-9661-aa96d47fccc0/resourceGroups/CreditSuisse2_GeorgeMoraitakis_ProjectExercise/overview) is empty.

We want you to create it in the softwireacademy directory so that:
- We can view it
- **You** don't have to pay for it

Reply:
WIll do it as part of module 12